pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -422,8 +422,12 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		reconciler.EmitEvent(c.Recorder, before, after, pr)
 		return err
 	}
-
-	candidateTasks, err := dag.GetSchedulable(d, pipelineState.SuccessfulPipelineTaskNames()...)
+	// Get the next tasks to be executed which doesnt't have any TaskRuns created yet
+	// ExecutedPipelineTaskNames gets the list of finished tasks (either successful or failed or skipped)
+	// GetSchedulable parses the whole graph and get a list of candidates based on already executed tasks
+	// e.g. for this Pipeline task1 -> task2 (runAfter task1) -> task3 (runAfter task2)
+	// GetSchedulable returns task1 followed by task2 (after task1 finishes) followed by task3 (after task2 finishes)
+	candidateTasks, err := dag.GetSchedulable(d, pipelineState.ExecutedPipelineTaskNames()...)
```

Comment:
This was little tricky to change, `SuccessfulPipelineTaskNames` returned list of tasks which were finished executing successfully. Now, with `runOn` failed tasks should also be included in the returned list.

Reply:
Perhaps `pipelineState.CompletedPipelineTaskNames()` ?

Reply:
thanks @dibyom for suggestion, replaced with `CompletedPipelineTaskNames`